examples/platform/silabs/display/lcd.cpp

```diff
@@ -126,7 +128,30 @@ int SilabsLCD::DrawPixel(void * pContext, int32_t x, int32_t y)
 
 int SilabsLCD::Update(void)
 {
-    return updateDisplay();
+    int status = 0;
+#ifdef SL_ENABLE_ICD_LCD
+    SilabsLCD::TurnOn();
+#endif // SL_ENABLE_ICD_LCD
+    status = updateDisplay();
+#ifdef SL_ENABLE_ICD_LCD
+    switch (mCurrentScreen)
+    {
+    case DemoScreen:
+        SilabsLCD::TurnOff(kActivityLCDTimeout);
+        break;
+    case StatusScreen:
+        SilabsLCD::TurnOff(kActivityLCDTimeout);
+        break;
+#ifdef QR_CODE_ENABLED
+    case QRCodeScreen:
+        SilabsLCD::TurnOff(kQRCodeScreenTimeout);
+        break;
+#endif // QR_CODE_ENABLED
+    default:
+        break;
+    }
+#endif // SL_ENABLE_ICD_LCD
+    return status;
```

Comment:
use #if rather than #ifdef and make sure the define always exist (1/0)
Also to limit the use of #if and simplify the code I suggest making one function that will turn on the screen and start the timer to turn it off and only use 1 Timeout value for all screen activities.
```suggestion
    #if SL_MATTER_TIMED_SCREEN_DISPLAY
    SilabsLCD::TimedScreenDisplay(kActivityLCDTimeout);
    #endif
    return updateDisplay();
```